include/simulation_time/Simulation_Time.hpp

```diff
@@ -68,6 +69,12 @@ class Simulation_Time
         total_output_times = simulation_total_time_seconds / output_interval_seconds + 1;
     }
 
+    Simulation_Time(const Simulation_Time& t, int interval) : Simulation_Time(t)
```

Comment:
This needs documentation of what it's doing. It looks like it's a copy constructor with an over-ridden output interval?

I would prefer if the logic of calculating `total_output_times` weren't duplicated from the other constructor.

More broadly, it seems like output interval shouldn't actually be a member of this at all, but I don't know the code well enough to be confident asserting that